ui/lib/core/addon/components/form-field.hbs

```diff
@@ -335,7 +336,7 @@
   {{else if (or (eq @attr.type "boolean") (eq @attr.options.editType "boolean"))}}
     <div class="b-checkbox">
       <input
-        disabled={{this.disabled}}
+        disabled={{and @attr.options.editDisabled (not @model.isNew)}}
```

Comment:
looking at the `disabled` getter in the [component file](https://github.com/hashicorp/vault/blob/main/ui/lib/core/addon/components/form-field.js#L95-L97) shows that a `@disabled` arg can be passed in. we still need to account for that case and this logic should be added to the getter 

We could then call `this.disabled` here and above. It might be worth updating the other areas of `and @attr.options.editDisabled (not @model.isNew)` so we're consistent (though as it is, not all inputs are even setup to accept a disabled arg...so the consistency is somewhat moot) This component file is always a challenge to balance keeping scope down without breaking backwards compatibility for an arg.